File: src/main/java/com/bank/service/ManagerService.java
package com.bank.service;

import com.bank.dto.ManagerAnalyticsResponse;
import com.bank.dto.TransactionResponse;

import java.util.List;

public interface ManagerService {
    List<TransactionResponse> getPendingApprovals();
    TransactionResponse processApproval(Long transactionId, boolean approve);
    ManagerAnalyticsResponse getAnalytics();
}

File: frontend/src/services/managerService.js
import API from './api';

export const managerService = {
  getPendingApprovals: () => API.get('/manager/approvals'),
  processApproval: (transactionId, approve) => API.post(`/manager/approvals/${transactionId}?approve=${approve}`),
  getAnalytics: () => API.get('/manager/analytics'),
};
